sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/CommunicationIdentityBuilderTests.java

```diff
@@ -106,6 +108,56 @@ public void buildAsyncClientTestUsingConnectionString() {
         assertNotNull(asyncClient);
     }
 
+    @Test
+    public void buildAsyncClientTestUsingConnectionStringAndClientOptions() {
+        builder
+            .connectionString(MOCK_CONNECTION_STRING)
+            .httpClient(new NoOpHttpClient() {
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    // It would be very difficult to test the actual header
```

Comment:
Good call on the HMAC, we have separate tests for them. I have a couple of suggestions though. These comments have their way of getting out of sync with the code. So I would recommend extracting the checks into a method and using that for both tests though. Calling such a method as explicit like `assertHMACHeadersExist`.
Also I do not think we need to check if UserAgent exist, I am pretty sure there are some tests in AzureCore for that